py/code_intelligence/triage.py

```diff
@@ -0,0 +1,518 @@
+"""Identify issues that need triage."""
+from code_intelligence import graphql
+from code_intelligence import util
+import fire
+import github3
+import json
+import logging
+import os
+import numpy as np
+import pprint
+import retrying
+import json
+
+TOKEN_NAME = "GITHUB_TOKEN"
+
+# TODO(jlewi): If we make this an app maybe we should read this from a .github
+#file
+ALLOWED_KINDS = ["improvement/enhancement", "community/question", "kind/bug"]
+ALLOWED_PRIORITY = ["priority/p0", "priority/p1", "priority/p2",
+                    "priority/p3"]
+
+REQUIRES_PROJECT = ["priority/p0", "priority/p1"]
+
+TRIAGE_PROJECT = "Needs Triage"
+
+# TODO(jlewi): Project card is currently hard coded
+# The notebook triage.ipynb contains a snippet to get the project card id
+PROJECT_CARD_ID = "MDEzOlByb2plY3RDb2x1bW41OTM0MzEz"
+
+class TriageInfo(object):
+  """Class describing whether an issue needs triage"""
+  def __init__(self):
+    self.issue = None
+    # Booleans indicating why triage might fail
+    self.missing_kind = True
+    self.missing_priority = True
+    self.missing_project = True
+    self.missing_area = True
+
+    self.triage_project_card = None
+
+  @classmethod
+  def from_issue(cls, issue):
+    """Construct TriageInfo from the supplied issue"""
+    info = TriageInfo()
+    info.issue = issue
+    labels = graphql.unpack_and_split_nodes(issue, ["labels", "edges"])
+
+    project_cards = graphql.unpack_and_split_nodes(issue,
+                                                   ["projectCards", "edges"])
+
+
+    for l in labels:
+      name = l["name"]
+
+      if name in ALLOWED_KINDS:
+        info.missing_kind = False
+
+      if name in ALLOWED_PRIORITY:
+        info.missing_priority = False
+
+        if not name in REQUIRES_PROJECT:
+          info.missing_project = False
+        else:
+          if project_cards:
+            info.missing_project = False
+
+      if name.startswith("area") or name.startswith("communit"):
```

Comment:
s/communit/community/ even though it works as is

Reply:
Done.